ARCHITECTURE.md

```diff
@@ -51,34 +51,40 @@ https://github.com/replicatedhq/kurl-api
-Additionally, this API interacts with the object store and is responsible for assembling and streaming air-gap bundles back to the end-user.
+The API for creating kURL URLs and rendering installers.
+The API accepts a kURL Installer spec and returns a URL including a deterministic hash of the spec for installing a cluster based on that spec, for example https://k8s.kurl.sh/5e61e80.
+The API interacts stores these hashes in its relational database for retrieval.
+Additionally, the API is responsible for resolving the add-on version ("latest" and ".x") and rendering the spec when a script is requested, for example https://k8s.kurl.sh/5e61e80/install.sh or shorthand https://k8s.kurl.sh/5e61e80.
+The API reads the current kURL version from the object storage bucket (https://kurl-sh.s3.amazonaws.com/dist/VERSION) and uses that version to lookup add-on version information (e.g. https://kurl-sh.s3.amazonaws.com/dist/v2022.04.19-0/supported-versions-gen.json) for spec resolution.
 
-#### Typescript API
+#### Go API Proxy
 
-The API for creating and consuming installers. Interacts with the relational database.
+This API proxies most requests directly to the Typescript API.
+Additionally, this API is responsible for assembling and streaming air-gap bundles back to the end-user.
+Air-gap packages are assembled from individual add-on package archives (e.g. https://kurl-sh.s3.amazonaws.com/dist/v2022.04.19-0/rook-1.0.4.tar.gz) and streamed as a single archive back to the end user when requested (e.g. https://k8s.kurl.sh/bundle/5e61e80.tar.gz).
 
 #### Relational Database
 
 Stores installer hashes.
 
 #### Object Store
 
-Stores built kURL assets including add-on bundles.
+Stores built kURL assets including add-on package archives.
 
 ### Architecture Diagram
 
-![kURL sh Architecture](https://user-images.githubusercontent.com/371319/166266866-267833a7-b21f-4665-b657-a55d6271b48b.png)
+![kURL.sh Architecture](https://user-images.githubusercontent.com/371319/166266866-267833a7-b21f-4665-b657-a55d6271b48b.png)
 
 ### Object Storage
 
 | Object | Description |
 | ------ | ----------- |
-| s3://[bucket]/[kURL version]/[entrypoint].tmpl           | Script templates rendered by the API. install.sh, join.sh, upgrade.sh and tasks.sh |
-| s3://[bucket]/[kURL version]/[addon-version].tar.gz      | Add-on bundles |
-| s3://[bucket]/[kURL version]/supported-versions-gen.json | Add-on and version information for a given release of kURL. For resolving "latest" and "dot x" versions as well as validating installers. |
+| s3://[bucket]/[(dist\|staging)]/VERSION                                    | The current kURL version, used by the API to determine what version subdirectory to serve packages from |
+| s3://[bucket]/[(dist\|staging)]/[kURL version]/supported-versions-gen.json | Add-on and version information for a given release of kURL, for resolving "latest" and "dot x" versions as well as validating installers |
```

Comment:
Who uses the supported-versions-gen.json? Where/how it is built? What is the project and code?

Reply:
https://github.com/replicatedhq/kURL/pull/3749/files#diff-8f6366fd8e7a5fa30f7f05879999195b7cf5fa1e3d157822eb37f0e91d226cfdR60